jam/server/api/v2/base.py

```diff
@@ -0,0 +1,296 @@
+import datetime
+import http.client
+
+import jwt
+
+import tornado.web
+
+from jam.auth import User
+from jam import exceptions
+from jam.auth import Permissions
+from jam.server.api.jsonapi import JSONAPIHandler
+
+ID_RE = r'[\d\w\.\-]{3,64}'
+ENDING = r'(?:/{})?/?'
+
+
+def ResourceEndpoint(view, serializer):
+    v1, v2, kwargs = [], [], {'view': view, 'serializer': serializer}
+    selector = r'(?P<{}_id>{})'.format(view.name, ID_RE)
+    for v in view.lineage()[:-1]:
```

Comment:
This class could use maybe some comments or something? It's a little hard to read and insane but it appears to get the job done. Also, I like the snowman smiley: `[:-1]`